runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java

```diff
@@ -47,21 +59,78 @@
 	@Inject
 	private OverlayManager overlayManager;
 
+	@Inject
+	private Client client;
+
+	@Inject
+	private StatusBarsConfig config;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean barsDisplayed;
+
+	private int lastCombatActionTickCount;
+
 	@Override
 	protected void startUp() throws Exception
 	{
+		clientThread.invokeLater(this::shouldToggleStatusBars);
 		overlayManager.add(overlay);
 	}
 
 	@Override
 	protected void shutDown() throws Exception
 	{
 		overlayManager.remove(overlay);
+		barsDisplayed = false;
 	}
 
 	@Provides
 	StatusBarsConfig provideConfig(ConfigManager configManager)
 	{
 		return configManager.getConfig(StatusBarsConfig.class);
 	}
+
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		shouldToggleStatusBars();
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (StatusBarsConfig.GROUP.equals(event.getGroup()) && event.getKey().equals("hideAfterCombatDelay"))
+		{
+			shouldToggleStatusBars();
+		}
+	}
+
+	private void shouldToggleStatusBars()
+	{
+		final Player localPlayer = client.getLocalPlayer();
+		if (localPlayer == null)
+		{
+			return;
+		}
+
+		final Actor interacting = localPlayer.getInteracting();
+
+		if ((interacting instanceof NPC && ArrayUtils.contains(((NPC) interacting).getComposition().getActions(), "Attack"))
+			|| (interacting instanceof Player && client.getVar(Varbits.PVP_SPEC_ORB) == 1))
+		{
+			lastCombatActionTickCount = client.getTickCount();
+			barsDisplayed = true;
+		}
+		else if (config.hideAfterCombatDelay() == 0)
```

Comment:
I think you can just check this above and do an early return if it is 0, since it isn't necessary to check anything else then?